Add and enhance features

chelaxian opened the pull request:
NEW

support of almost all models in 1minAI API
support of voice models (TTS and STT)
support of IMAGE_GENERATOR models (also works from /v1/chat/conversations route)
support of IMAGE_VARIATOR models
support of files (assets) upload
support of webSearch / site crawling and function_callings (tools)
support of CHAT_WITH_IMAGE and CHAT_WITH_PDF

chelaxian committed:
Update INFO.md

chelaxian committed:
Update INFO.md

chelaxian committed:
Update INFO.md

chelaxian committed:
Update __init__.py

chelaxian committed:
Update audio.py

chelaxian committed:
Update files.py

chelaxian committed:
Update images.py

chelaxian committed:
Update text.py

chelaxian committed:
Update __init__.py

chelaxian committed:
Update common.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update imports.py

chelaxian committed:
Update logger.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update app.py

chelaxian committed:
Create utils.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update text.py

chelaxian committed:
Update images.py

chelaxian committed:
Update text.py

chelaxian committed:
Update images.py

chelaxian committed:
Update text.py

chelaxian committed:
Update images.py

chelaxian committed:
Update text.py

chelaxian committed:
Update text.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update images.py

chelaxian committed:
Update text.py

chelaxian committed:
Update text.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update README.md

chelaxian committed:
Update and rename INFO.md to CODE_STRUCTURE.md

chelaxian committed:
Merge pull request #1 from chelaxian/test

Test

chelaxian committed:
Create UPDATE.sh

chelaxian committed:
Update README.md

chelaxian committed:
Update README.md

chelaxian committed:
Update docker-compose.yml

chelaxian committed:
Update UPDATE.sh

kokofixcomputers commented:
This project changes some readme urls. Can you revert them? I need them to still point to my repo. Thanks!

chelaxian commented:
> This project changes some readme urls. Can you revert them? I need them to still point to my repo. Thanks!

I moved old monolith code to backup/old/main.py

I decide to restructure code to module architecture (in process)

Old code is in backup-main tree

I think the project is too dirty to merge to your clean and optimized code. It has too many code block duplicates, unnecessary functions on so on. I need to modify many parts

chelaxian commented:
> This project changes some readme urls. Can you revert them? I need them to still point to my repo. Thanks!

I realized that this fork not universal to use, but adopted to certain telegram bot. It mostly converts specific models to use single entry point v1/chat/conversations instead of files/speech/image and so on. It is ok for clients that support only text-only entrypoints, so many parts of code are unnecessary for most users

chelaxian committed:
Update README.md

chelaxian committed:
Update README.md

chelaxian committed:
Update README.md

chelaxian committed:
Create INSTALL.sh

chelaxian committed:
Create RUN.sh

chelaxian committed:
Update README.md

chelaxian committed:
Update README.md

chelaxian committed:
Update README.md

chelaxian committed:
Update README.md

chelaxian committed:
Update README.md

chelaxian committed:
Update README.md

chelaxian committed:
Update UPDATE.sh

chelaxian committed:
Update constants.py

chelaxian committed:
Update images.py

chelaxian committed:
Create STOP.sh

chelaxian committed:
Update README.md

chelaxian committed:
Update __init__.py

chelaxian committed:
Update audio.py

chelaxian committed:
Update files.py

chelaxian committed:
Update images.py

chelaxian committed:
Update text.py

chelaxian committed:
Update and rename utils.py to functions.py

chelaxian committed:
Update README.md

chelaxian committed:
Update functions.py

chelaxian committed:
Update text.py

chelaxian committed:
Update text.py

chelaxian committed:
Update functions.py

chelaxian committed:
Rename routes/STOP.sh to STOP.sh

chelaxian committed:
Create __init__.py

chelaxian committed:
Create audio_func.py

chelaxian committed:
Create file_func.py

chelaxian committed:
Create img_func.py

chelaxian committed:
Create shared_func.py

chelaxian committed:
Create txt_func.py

chelaxian committed:
Update __init__.py

chelaxian committed:
Update CODE_STRUCTURE.md

chelaxian committed:
Update README.md

chelaxian committed:
Rename docker-compose.yml to backup/old/docker-compose.yml

chelaxian committed:
Create README_EN.md

chelaxian committed:
Merge pull request #2 from chelaxian/test3

Test3

chelaxian committed:
Create CODE_STRUCTURE_EN.md

chelaxian committed:
Update CODE_STRUCTURE_EN.md

chelaxian committed:
Update CODE_STRUCTURE.md

chelaxian committed:
Update CODE_STRUCTURE_EN.md

chelaxian committed:
Update CODE_STRUCTURE.md

chelaxian committed:
Update CODE_STRUCTURE_EN.md

chelaxian committed:
Update UPDATE.sh

chelaxian committed:
Update INSTALL.sh

chelaxian committed:
Update text.py

chelaxian committed:
Update txt_func.py

kokofixcomputers commented:
> shell i close this pull request at this point? or not?

Sure... It would be helpful if you could also create a branch that adds some Image generation and other features without changing the language and other things.

If not, can I just take some of your code that adds these things?

chelaxian committed:
Update images.py

chelaxian committed:
Update img_func.py

chelaxian commented:
> > shell i close this pull request at this point? or not?
> 
> Sure... It would be helpful if you could also create a branch that adds some Image generation and other features without changing the language and other things and merge that branch into my repo, that would be helpful, considering you've already done all this work. (basically making another pr)
> 
> If not, can I just take some of your code that adds these things?

i plan to translate code after it will be fully refactored and optimized
also now i translate README_EN.md and CODE_CTRUCTURE_EN.md

i splited main.py onto 16 separate modules to simplify code review

chelaxian committed:
Merge pull request #3 from chelaxian/test4

Test4

chelaxian committed:
Update audio.py

chelaxian committed:
Update audio_func.py

chelaxian committed:
Update files.py

chelaxian committed:
Update file_func.py

chelaxian committed:
Update img_func.py

chelaxian committed:
Update shared_func.py

chelaxian committed:
Update audio_func.py

chelaxian committed:
Update README.md

chelaxian committed:
Update README_EN.md

chelaxian committed:
Update images.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update audio.py

chelaxian committed:
Update __init__.py

chelaxian committed:
Update CODE_STRUCTURE.md

chelaxian committed:
Update CODE_STRUCTURE_EN.md

chelaxian committed:
Update __init__.py

chelaxian committed:
Update __init__.py

chelaxian committed:
Update functions.py

chelaxian committed:
Update functions.py

chelaxian committed:
Update __init__.py

chelaxian committed:
Update functions.py

chelaxian committed:
Update __init__.py

chelaxian committed:
Update images.py

chelaxian committed:
Update CODE_STRUCTURE_EN.md

chelaxian committed:
Update CODE_STRUCTURE.md

chelaxian committed:
Update images.py

chelaxian committed:
Update __init__.py

chelaxian committed:
Update functions.py

chelaxian committed:
Update text.py

chelaxian committed:
Update audio.py

chelaxian committed:
Update images.py

chelaxian committed:
Update CODE_STRUCTURE_EN.md

chelaxian committed:
Update CODE_STRUCTURE.md

chelaxian committed:
Update shared_func.py

chelaxian committed:
Update common.py

chelaxian committed:
Update imports.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Create async_tasks.py

chelaxian committed:
Delete routes/async_tasks.py

chelaxian committed:
Update img_func.py

chelaxian committed:
Update audio_func.py

chelaxian committed:
Merge pull request #4 from chelaxian/test6

Test6

chelaxian committed:
Update README_EN.md

chelaxian committed:
Update README.md

chelaxian committed:
Update text.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update txt_func.py

chelaxian committed:
Update app.py

chelaxian committed:
Update __init__.py

chelaxian committed:
Update common.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update imports.py

chelaxian committed:
Update logger.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update app.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update img_func.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update app.py

chelaxian committed:
Update app.py

chelaxian committed:
Update memcached.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update imports.py

chelaxian committed:
Merge pull request #6 from chelaxian/new1

New1

chelaxian committed:
Update audio_func.py

chelaxian committed:
Update audio.py

chelaxian committed:
Update text.py

chelaxian committed:
Merge pull request #7 from chelaxian/new1

New1

chelaxian committed:
Update constants.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update constants.py

chelaxian committed:
Update constants.py

nbmorgan commented:
This is a test

chelaxian committed:
Fix 1min.ai streaming 406/404 issues

Inject browser-like headers for api.1min.ai requests and use ?isStreaming=true with SSE Accept header for streaming.

Made-with: Cursor

chelaxian committed:
Update model allowlists in constants

Sync 1min-relay model lists with uni-api config (GPT-5.4/5.2, Gemini 3.1 previews, Claude 4.6, Qwen, and new image model IDs).

Made-with: Cursor

chelaxian committed:
Fix 1min.ai streaming crawl-prefix and image variations

Parse 1min.ai SSE properly and drop leading '🌐 Crawling site' trace lines without disabling streaming/tools. Also fix image variation upload MIME/extension detection.

Made-with: Cursor

chelaxian committed:
Fix: strip 1min.ai crawl/tool prefix from replies

Remove leading 'Crawling site' and tool scaffolding (read/memory_search) from both streamed SSE and non-streamed responses.

chelaxian committed:
Update relay for 1min.ai unified chat API

Switch chat proxying to the new Chat with AI API (UNIFY_CHAT_WITH_AI), improve SSE handling, and refresh docs. Add a note to the bundled legacy API doc pointing to the official documentation.

Made-with: Cursor

chelaxian committed:
Deps: enable Dependabot for pip

Configure Dependabot to check requirements.txt weekly and open dependency update PRs.

Made-with: Cursor

chelaxian committed:
Update dependabot.yml

dependabot[bot] committed:
deps(deps): bump mistral-common from 1.5.4 to 1.10.0

Bumps [mistral-common](https://github.com/mistralai/mistral-common) from 1.5.4 to 1.10.0.
- [Release notes](https://github.com/mistralai/mistral-common/releases)
- [Commits](https://github.com/mistralai/mistral-common/compare/v1.5.4...v1.10.0)

---
updated-dependencies:
- dependency-name: mistral-common
  dependency-version: 1.10.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
deps(deps): bump python-dotenv from 1.0.1 to 1.2.2

Bumps [python-dotenv](https://github.com/theskumar/python-dotenv) from 1.0.1 to 1.2.2.
- [Release notes](https://github.com/theskumar/python-dotenv/releases)
- [Changelog](https://github.com/theskumar/python-dotenv/blob/main/CHANGELOG.md)
- [Commits](https://github.com/theskumar/python-dotenv/compare/v1.0.1...v1.2.2)

---
updated-dependencies:
- dependency-name: python-dotenv
  dependency-version: 1.2.2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
deps(deps): bump flask-limiter from 3.11.0 to 4.1.1

Bumps [flask-limiter](https://github.com/alisaifee/flask-limiter) from 3.11.0 to 4.1.1.
- [Release notes](https://github.com/alisaifee/flask-limiter/releases)
- [Changelog](https://github.com/alisaifee/flask-limiter/blob/master/HISTORY.rst)
- [Commits](https://github.com/alisaifee/flask-limiter/compare/3.11.0...4.1.1)

---
updated-dependencies:
- dependency-name: flask-limiter
  dependency-version: 4.1.1
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

chelaxian committed:
Create SECURITY.md

chelaxian committed:
Merge pull request #8 from chelaxian/dependabot/pip/mistral-common-1.10.0

deps(deps): bump mistral-common from 1.5.4 to 1.10.0

chelaxian committed:
Merge pull request #9 from chelaxian/dependabot/pip/python-dotenv-1.2.2

deps(deps): bump python-dotenv from 1.0.1 to 1.2.2

chelaxian committed:
Merge branch 'main' into dependabot/pip/flask-limiter-4.1.1

dependabot[bot] committed:
deps(deps): bump requests from 2.32.3 to 2.33.1

Bumps [requests](https://github.com/psf/requests) from 2.32.3 to 2.33.1.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.32.3...v2.33.1)

---
updated-dependencies:
- dependency-name: requests
  dependency-version: 2.33.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

chelaxian committed:
Merge pull request #10 from chelaxian/dependabot/pip/flask-limiter-4.1.1

deps(deps): bump flask-limiter from 3.11.0 to 4.1.1

dependabot[bot] committed:
deps(deps): bump flask from 3.1.0 to 3.1.3

Bumps [flask](https://github.com/pallets/flask) from 3.1.0 to 3.1.3.
- [Release notes](https://github.com/pallets/flask/releases)
- [Changelog](https://github.com/pallets/flask/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/flask/compare/3.1.0...3.1.3)

---
updated-dependencies:
- dependency-name: flask
  dependency-version: 3.1.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

chelaxian committed:
Merge pull request #11 from chelaxian/dependabot/pip/requests-2.33.1

deps(deps): bump requests from 2.32.3 to 2.33.1

chelaxian committed:
Merge pull request #12 from chelaxian/dependabot/pip/flask-3.1.3

deps(deps): bump flask from 3.1.0 to 3.1.3

dependabot[bot] committed:
deps(deps): bump limits from 4.4.1 to 5.8.0

Bumps [limits](https://github.com/alisaifee/limits) from 4.4.1 to 5.8.0.
- [Release notes](https://github.com/alisaifee/limits/releases)
- [Changelog](https://github.com/alisaifee/limits/blob/master/HISTORY.rst)
- [Commits](https://github.com/alisaifee/limits/compare/4.4.1...5.8.0)

---
updated-dependencies:
- dependency-name: limits
  dependency-version: 5.8.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

chelaxian committed:
Merge pull request #14 from chelaxian/dependabot/pip/limits-5.8.0

deps(deps): bump limits from 4.4.1 to 5.8.0

dependabot[bot] committed:
deps(deps): bump flask-cors in the pip group across 1 directory

Bumps the pip group with 1 update in the / directory: [flask-cors](https://github.com/corydolphin/flask-cors).


Updates `flask-cors` from 4.0.0 to 6.0.0
- [Release notes](https://github.com/corydolphin/flask-cors/releases)
- [Changelog](https://github.com/corydolphin/flask-cors/blob/main/CHANGELOG.md)
- [Commits](https://github.com/corydolphin/flask-cors/compare/4.0.0...6.0.0)

---
updated-dependencies:
- dependency-name: flask-cors
  dependency-version: 6.0.0
  dependency-type: direct:production
  dependency-group: pip
...

Signed-off-by: dependabot[bot] <support@github.com>

chelaxian committed:
Merge pull request #17 from chelaxian/dependabot/pip/pip-8fb5dba437

deps(deps): bump flask-cors from 4.0.0 to 6.0.0 in the pip group across 1 directory

dependabot[bot] committed:
deps(deps): bump flask-cors from 4.0.0 to 6.0.2

Bumps [flask-cors](https://github.com/corydolphin/flask-cors) from 4.0.0 to 6.0.2.
- [Release notes](https://github.com/corydolphin/flask-cors/releases)
- [Changelog](https://github.com/corydolphin/flask-cors/blob/main/CHANGELOG.md)
- [Commits](https://github.com/corydolphin/flask-cors/compare/4.0.0...6.0.2)

---
updated-dependencies:
- dependency-name: flask-cors
  dependency-version: 6.0.2
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

chelaxian committed:
Merge branch 'main' into dependabot/pip/flask-cors-6.0.2

chelaxian committed:
Merge pull request #13 from chelaxian/dependabot/pip/flask-cors-6.0.2

deps(deps): bump flask-cors from 4.0.0 to 6.0.2

chelaxian committed:
Fix: ignore 1min.ai SSE result events

Do not forward Chat with AI SSE event=result (aiRecord JSON) into OpenAI-like streamed content.

Made-with: Cursor

chelaxian committed:
Update model lists, feature routing, and usage mapping

Align model allow-lists with latest 1min.ai docs, support AI Feature API types (e.g. CODE_GENERATOR), and map aiRecord.metadata tokens into OpenAI-compatible usage for both streaming and non-streaming responses.

Made-with: Cursor

chelaxian committed:
Improve vision allow-lists and files content support

Split vision allow-lists for Chat-with-Image vs UNIFY chat, add /v1/files/{id}/content streaming via stored asset metadata, and pass through code_interpreter/function tools without breaking OpenAI clients.

Made-with: Cursor

chelaxian committed:
Improve image model coverage and restrict variations

Add payload support for additional image generator models, and make image variations available only for models that support both generation and variation (no fallback substitutions).

Made-with: Cursor

chelaxian committed:
Align image payloads with 1min.ai docs

Map Midjourney image models to Magic Art, add Dzine default-style auto-pick via /api/dzine/styles, and update Magic Art + Stable Image Core payload fields to match documented parameters.

Made-with: Cursor

chelaxian committed:
Enable OpenClaw-style tool calling via JSON emulation.

- Add tool/function roles to conversation history and parse JSON tool_calls into OpenAI-compatible responses.
- Expand function-calling model allowlist and disable streaming when function tools are requested.

Made-with: Cursor

chelaxian committed:
Ignore local comparison repo folder.

Made-with: Cursor

chelaxian committed:
Add OpenAI /v1/responses and dynamic model registry.

Also retry once without web_search on upstream 400 and expose X-WebSearch-Degraded.

Made-with: Cursor

chelaxian committed:
Align UNIFY_CHAT_WITH_AI settings with OpenClaw 1min plugin.

Always include webSearchSettings/withMemories and fix tool-instructions prompt_text ordering.

Made-with: Cursor

chelaxian committed:
Move 1min.ai API docs into backup folder.

Made-with: Cursor

chelaxian committed:
Document responses API, dynamic models, and web_search fallback.

Made-with: Cursor

chelaxian committed:
Fix: keep streaming even with function tools

Do not force non-streaming when requests include function tools, since many clients always send them. This restores real-time SSE streaming behavior.

Made-with: Cursor

chelaxian committed:
Gate tool-calling to OpenClaw and preserve streaming

Ignore tool definitions from non-OpenClaw clients to avoid breaking SSE streaming. For OpenClaw, probe via a single non-stream upstream call and only disable streaming when the model returns tool_calls.

Made-with: Cursor

chelaxian committed:
Fix robust text extraction from 1min.ai responses

Improve tool-calling reliability by extracting assistant content from multiple 1min.ai response shapes, preventing empty completions that block OpenClaw memory/file writes.

Made-with: Cursor

chelaxian committed:
Docs: describe OpenClaw tool-calling emulation

Update READMEs with OpenClaw-only tools gating, probe behavior, and robust upstream text extraction to prevent empty completions.

Made-with: Cursor

chelaxian committed:
Fix: preserve OpenClaw tool-calls before stripping traces

Extract tool_calls from raw upstream text first in OpenClaw mode so tool traces aren't removed by crawl/tool noise stripping, preventing silent replies and enabling memory/file writes.

Made-with: Cursor

chelaxian committed:
Fix: parse tool-trace outputs into OpenAI tool_calls

Support provider outputs like `tool\nwrite\n{...}` by converting them into OpenAI-style `tool_calls`, preventing empty completions and enabling OpenClaw memory/file writes.

Made-with: Cursor

chelaxian committed:
Fix: OpenClaw probe-empty fallback to upstream streaming

When OpenClaw tool-mode probe returns empty content, retry via upstream streaming and collect full content to detect tool traces/tool_calls, preventing silent replies.

Made-with: Cursor

chelaxian committed:
Fix: OpenClaw non-stream empty fallback to upstream SSE

If an OpenClaw tools request returns empty content in non-stream mode, retry via upstream streaming and transform the final result/content into OpenAI-like output.

Made-with: Cursor

chelaxian committed:
Fix: treat whitespace-only content as empty in OpenClaw fallbacks

Trigger upstream SSE fallbacks when probe/non-stream returns only whitespace, and prefer parsing event:result JSON when available.

Made-with: Cursor

chelaxian committed:
Fix: detect zero-width/whitespace-only assistant content

Treat common zero-width characters as empty so OpenClaw fallbacks trigger reliably when upstream returns invisible output.

Made-with: Cursor

chelaxian committed:
Fix: synthesize deterministic memory write tool_calls for OpenClaw

Bypass flaky upstream tool-mode outputs for "обнови свою память" by returning OpenAI tool_calls directly (write MEMORY.md and memory/YYYY-MM-DD.md).

Made-with: Cursor

chelaxian committed:
Fix: repair non-stream memory tool_calls syntax

Remove broken escaped quotes in deterministic OpenClaw memory write tool_calls for non-stream requests.

Made-with: Cursor

chelaxian committed:
Refactor: remove OpenClaw memory trigger hack

Drop deterministic phrase-based tool_call synthesis and rely on generic OpenClaw tool-call probe flow; add a probe timeout to avoid hanging.

Made-with: Cursor

chelaxian committed:
Fix: force OpenClaw stream fallback on zero output tokens

When OpenClaw tool-probe non-stream response has outputToken==0 and no tool_calls, force upstream streaming fallback to avoid silent empty replies.

Made-with: Cursor

chelaxian committed:
Fix: force OpenClaw stream fallback on zero completion tokens

Some probe responses produce completion_tokens=0 even when upstream metadata is missing/ambiguous; treat this as a signal to run streaming fallback.

Made-with: Cursor

chelaxian committed:
Debug: log OpenClaw streaming fallback triggers

Add INFO logs when probe returns 0 tokens and when upstream streaming fallback starts, to diagnose silent memory updates.

Made-with: Cursor

chelaxian committed:
Fix: never return silent empty reply to OpenClaw

If both non-stream probe and streaming fallback yield empty content and no tool_calls, emit a non-empty assistant error message so Telegram/OpenClaw never appear "silent".

Made-with: Cursor

chelaxian committed:
feat: openclaw-bridge на loopback, relay без OpenClaw tool-mode

Добавлен openclaw-bridge (FastAPI): OpenAI /v1/chat/completions с tools, upstream к 1min-relay, парсинг tool_calls.
Удалены probe/streaming fallback и режим openclaw_tool_mode из relay; tools не форвардятся в 1min.ai.
Обновлены README, примеры deploy (systemd, env) и unit-тесты парсера.

Made-with: Cursor

chelaxian committed:
fix: bridge fallback на stream и компактные tool инструкции

Уменьшены tool-инструкции для снижения prompt size.
Добавлен retry через upstream stream:true при пустом non-stream ответе.
Дополнена документация по установке/запуску bridge.

Made-with: Cursor

chelaxian committed:
fix: stream tool_calls в bridge для OpenClaw

Made-with: Cursor

chelaxian committed:
fix: валидировать tool_calls и 1 retry при ошибке

Made-with: Cursor

chelaxian committed:
feat: loop-guard в bridge против токен-дрейна

Made-with: Cursor

chelaxian committed:
chore: усилить инструкцию edit для точного oldText

Made-with: Cursor

chelaxian committed:
fix: loop-guard возвращает 200 чтобы OpenClaw не уходил в cooldown

Made-with: Cursor

chelaxian committed:
fix: принудительный retry tool_calls для MEMORY.md (fallback model)

Made-with: Cursor

chelaxian committed:
fix: форс tool retry только для update MEMORY.md

Made-with: Cursor

chelaxian committed:
fix: fallback tool_calls для чтения/обновления MEMORY.md

Made-with: Cursor

chelaxian committed:
fix: forced tool retry требует non-empty tool_calls

Made-with: Cursor

chelaxian committed:
fix: универсальный strict tools contract (без спецкейсов)

Made-with: Cursor

chelaxian committed:
docs: обновить описание openclaw-bridge и strict tools contract

Made-with: Cursor

chelaxian committed:
fix: strict tools contract также для запросов чтения/доказательств

Made-with: Cursor

chelaxian committed:
fix: парсинг fn(args) + hard-fail если strict contract без tool_calls

Made-with: Cursor

chelaxian committed:
fix: strict tools contract также для запросов изменения артефактов

Made-with: Cursor

chelaxian committed:
feat: опционально форсить модель для tool-calling

Made-with: Cursor

chelaxian committed:
chore: debug strict tools contract logging

Made-with: Cursor

chelaxian committed:
fix: локально отдавать MEMORY.md цитатой (без tool_calls)

Made-with: Cursor

chelaxian committed:
fix: Magic Art вариации + таймауты fast/relax

Добавлен прямой путь IMAGE_VARIATOR для magic-art* при командах [_V1_]..[_V4_], чтобы не запускалась генерация заново. Также введены раздельные таймауты image-операций для fast/relax и минимальный unittest на регрессию.

Made-with: Cursor

chelaxian committed:
fix: вариатор Magic Art без истории + aspect clamp

Сохраняем последние сгенерированные URLs по api_key+model и используем их при командах [_V1_]..[_V4_] когда бот не передаёт историю. Для Magic Art вариатора нормализуем aspect к поддерживаемым значениям по документации 1min.ai.

Made-with: Cursor

chelaxian committed:
fix: нормализация image моделей и фоллбэк вариатора

Добавлен устойчивый маппинг UI-имен моделей (генератор/вариатор) и выбор вариаторной модели с переходом на Flux Redux Schnell при несовпадении названий.

Made-with: Cursor

chelaxian committed:
fix: фоллбэк вариатора по семействам моделей

При запросе вариаций теперь выбирается ближайший variator по вендору/семейству (Flux→Flux Redux, MJ/MagicArt→MagicArt, OpenAI images→dall-e-2) и общий фоллбэк на flux-redux-schnell.

Made-with: Cursor

chelaxian committed:
fix: разрешить flux-redux вариатор через VARIATION_SUPPORTED_MODELS

Для выбора модели вариатора используем список поддерживаемых вариаторов (VARIATION_SUPPORTED_MODELS), а не пересечение с генераторами, чтобы flux-redux-* всегда были доступны как фоллбэк.

Made-with: Cursor

chelaxian committed:
docs: зеркало skill dev-github (маршрутизация Cursor/Claude, heartbeat, gh auth)

Made-with: Cursor

chelaxian committed:
feat(openclaw-skills): cursor-dispatch wrapper and orchestrator rules for dev-github

Made-with: Cursor
